### PR TITLE
Expand/Collapse button for Job Result JSON viewer

### LINKED ISSUE #959: [React New Frontend] provide navigable JSON result for each analyzer
In the IntelOwl v3 version we have this navigable component that allows us to easily surf the JSON result of an analyzers

image

In IntelOwl v4, by default, we get all the JSON data at the same time. While this can be helpful, most of the time is not.
Example:
image

My idea is to replace that new component with something similar to the previous one available in the Angular GUI

### PR DESCRIPTION
**Pull request by devmrfitz**
# Description

Added expand/collapse button in Job result JSON viewer

## Related issues
#959 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [ ] I have read and understood the rules about [how to Contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) to this project
- [ ] The pull request is for the branch `develop`
- [ ] A new analyzer or connector was added, in which case:
    - [ ] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
    - [ ] [Advanced-Usage](./Advanced-Usage.md) was updated (in case the analyzer/connector provides additional optional configuration).
    - [ ] Secrets were added in [env_file_app_template](https://github.com/intelowlproject/IntelOwl/blob/master/docker/env_file_app_template), [env_file_app_ci](https://github.com/certego/IntelOwl/blob/master/docker/env_file_app_ci) and in the [Installation](./Installation.md) docs, if necessary.
    - [ ] If the analyzer/connector requires mocked testing, `_monkeypatch()` was used in its class to apply the necessary decorators.
    - [ ] If a File analyzer was added, its name was explicitly defined in [test_file_scripts.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_file_scripts.py) (not required for Observable Analyzers).
- [ ] If external libraries/packages with restrictive licenses were used, they were added in the [Legal Notice](https://github.com/certego/IntelOwl/blob/master/.github/legal_notice.md) section.
- [ ] The tests gave 0 errors.
- [ ] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [ ] The commits were squashed into a single one (optional, they will be squashed anyway by the maintainer)
- [ ] If changes were made to an existing model/serializer/view, the docs were updated and regenerated (check `CONTRIBUTE.md`).
  
### Important Rules
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review

# Real World Example

Please delete if the PR is for bug fixing.
Otherwise, please provide the resulting raw JSON of a finished analysis (and, if you like, a screenshot of the results). This is to allow the maintainers to understand how the analyzer works.

### CODE REVIEW

**Comment by devmrfitz** [comment]
image

File: frontend/src/components/jobs/result/utils/tables.jsx [hunk 1]
```diff
@@ -106,20 +108,32 @@ const tableProps = {
...
-      style={{ height: "50vh", width: "90vw", overflow: "scroll" }}
-    />
-  ),
+  SubComponent: ({ row }) => {
+    const [collapsed, setCollapsed] = React.useState(1);
+    return (
+      <div>
+        <Button
+          onClick={() => {
+            setCollapsed(collapsed ? false : 1);
```
Comment (drosetti) [inline frontend/src/components/jobs/result/utils/tables.jsx:117]: I think this condition does not manage all the cases

**Review by drosetti — CHANGES_REQUESTED** [review]
Good, but there are two bugs:
* If you open and then close the "report" key the toggle button stops working
* If you open "report" key and then "resolutions" key the toggle button stops working

**Comment by devmrfitz** [comment]
>     * If you open and then close the "report" key the toggle button stops working
> 
>     * If you open "report" key and then "resolutions" key the toggle button stops working

This is the intended behaviour of `react-json-view`. When we click on "report", we set an individual collapse depth for it, hence, it does not respect the overall depth. The depth is still respected by other keys, for instance, Runtime Configuration, as seen below

image
image

**Comment by devmrfitz** [comment]
> > ```
> > * If you open and then close the "report" key the toggle button stops working
> > 
> > * If you open "report" key and then "resolutions" key the toggle button stops working
> > ```
> 
> This is the intended behaviour of `react-json-view`. When we click on "report", we set an individual collapse depth for it, hence, it does not respect the overall depth. The depth is still respected by other keys, for instance, Runtime Configuration, as seen below
> 
> image image

@drosetti

**Comment by drosetti** [comment]
Ok, but this makes the button not working, and it feels like a bug or an unexpected behaviour from an user perspective. I think the buttons always as to hide/show all the elements, but the arrow buttons have to show/hide the single value of the dict.

**Comment by devmrfitz** [comment]
> Ok, but this makes the button not working, and it feels like a bug or an unexpected behaviour from an user perspective. I think the buttons always as to hide/show all the elements, but the arrow buttons have to show/hide the single value of the dict.

I tried all combinations at the [demo site](https://mac-s-g.github.io/react-json-view/demo/dist/), but this behaviour does not seem to be controllable in any manner

**Comment by drosetti** [comment]
> > Ok, but this makes the button not working, and it feels like a bug or an unexpected behaviour from an user perspective. I think the buttons always as to hide/show all the elements, but the arrow buttons have to show/hide the single value of the dict.
> 
> I tried all combinations at the [demo site](https://mac-s-g.github.io/react-json-view/demo/dist/), but this behaviour does not seem to be controllable in any manner

Damn, you're right.
I think  the library behaviour is non-intuitive. I close this PR and write a comment about `collapsed` in the issue. Thanks for you tests!

**Pull request closed without merging**